lib/adyen/rest/client.rb

```diff
@@ -19,15 +20,14 @@ module REST
     class Client
       include AuthorisePayment
       include ModifyPayment
+      include Payout
 
       attr_reader :environment
 
       # @param environment [String] The Adyen environment to interface with. Either
       #   <tt>'live'</tt> or <tt>'test'</tt>.
-      # @param username [String] The webservice username, e.g. <tt>ws@Company.Account</tt>
-      # @param password [String] The password associated with the username
-      def initialize(environment, username, password)
-        @environment, @username, @password = environment, username, password
+      def initialize(environment)
+        @environment = environment
```

Comment:
This breaks the encapsulation property of the `Client` class. Because the client class now will have to talk to `Adyen.configuration`, and `Adyen.configuration` is a singleton, it automatically but unnecessarily makes `Adyen::Client` a singleton too. This means it's hard to have two different clients in a single application, and potentially will have thread safety issues.

What about this:

``` ruby
  def initialize(environment, deprecated_username = nil, deprecated_password = nil, api_username: nil, api_password: nil, payout_store_username: nil, payout_store_password: nil, ...)
    
    @environment = environment
    @api_username, @api_password = api_username || deprecated_username, api_password || deprecated_password
    @payout_store_username, @payout_store_password = payout_store_username, payout_store_password
    ...
```

This maintains backwards compatibility, and removes the dependency on `Adyen.configuration`